Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -627,6 +627,37 @@ void LocalFrameViewLayoutContext::popLayoutState()
     }
 }
 
+
+bool LocalFrameViewLayoutContext::markNeedingTransformUpdate(RenderLayerModelObject* containingBlock, RenderLayerModelObject* descendant)
+{
+    if (!containingBlock)
+        return false;
+    auto it = m_containersWithDescendantsNeedingTransformUpdate.find(*containingBlock);
+    if (it != m_containersWithDescendantsNeedingTransformUpdate.end()) {
```

Comment:
I still wonder if this can be written more efficiently with `ensure()`

Reply:
If not, you can still use the iterator returned by `find` for insertion.

Reply:
Ah right I misinterpreted the point of `ensure()`.